http/vibe/http/common.d

```diff
@@ -712,6 +712,45 @@ final class Cookie {
 		}
 	}
 
+	/// Parses the cookie from a header field, returning the name of the cookie.
+	string parse(string headerString) {
```

Comment:
I think it's really confusing that parse returns a string and alters the internal state, wouldn't it be smarter to make this a static method and return a `Cookie` instance?

Reply:
A tuple of `string` and `Cookie` that is, but I agree that this would be nicer. Maybe this method should just be `package` for now, so that we can move on with the main functionality without worriying too much about the API here (which may have to change in the future anyway when `Cookie` for example gets changed to a `struct`).

Reply:
From what I have observed, most parsing functions are free-standing, so it makes more sense to make `name` part of `Cookie` class and have a `parseHTTPCookie` free-standing function, which returns a cookie